src/Blocks/BlockBed.cpp

```diff
@@ -145,4 +159,72 @@ bool cBlockBedHandler::OnUse(cChunkInterface & a_ChunkInterface, cWorldInterface
 
 
 
+void cBlockBedHandler::OnPlacedByPlayer(cChunkInterface & a_ChunkInterface, cWorldInterface & a_WorldInterface, cPlayer * a_Player, const sSetBlock & a_BlockChange)
+{
+	bool IsHead = false;
+	if (a_BlockChange.m_BlockMeta & 0x08)
+	{
+		IsHead = true;
+	}
+
+	class cBedColor :
+		public cBedCallback
+	{
+	public:
+		short m_Color;
+		bool m_IsHead;
+
+		cBedColor(short a_Color, bool a_IsHead) :
+			m_Color(a_Color),
+			m_IsHead(a_IsHead)
+		{
+		}
+
+		virtual bool Item(cBedEntity * a_Bed) override
+		{
+			a_Bed->SetColor(m_Color);
+			a_Bed->SetIsHead(m_IsHead);
+			return true;
+		}
+	};
+	cBedColor BedCallback(a_Player->GetEquippedItem().m_ItemDamage, IsHead);
+	a_Player->GetWorld()->DoWithBedAt(a_BlockChange.GetX(), a_BlockChange.GetY(), a_BlockChange.GetZ(), BedCallback);
+
+	int posX = a_BlockChange.GetX();
+	int posY = a_BlockChange.GetY();
+	int posZ = a_BlockChange.GetZ();
+
+	// If the bed entity is send immediately, the client (maybe) still has not the bed.
+	// Fix that by delaying the broadcast of the bed entity by a tick:
+	a_Player->GetWorld()->ScheduleTask(1, [posX, posY, posZ](cWorld & a_World)
+	{
+		a_World.BroadcastBlockEntity(posX, posY, posZ);
+	});
+}
+
+
+
+
+
+void cBlockBedHandler::ConvertToPickups(cEntity * a_Digger, cItems & a_Pickups, NIBBLETYPE a_BlockMeta, int a_BlockX, int a_BlockY, int a_BlockZ)
+{
+	class cBedColor :
+		public cBedCallback
+	{
+	public:
+		short m_Color;
+
+		cBedColor()
+		{
```

Comment:
I think you should initialize `m_Color` here with a default value, just in case the callback fails for some reason.